documentation/library_guide/kernel_templates/single_pass_scan.rst

```diff
@@ -0,0 +1,208 @@
+| Name                                          | Description                                                         |
++===============================================+=====================================================================+
+|  ``q``                                        | The SYCL* queue where kernels are submitted.                        |
++-----------------------------------------------+---------------------------------------------------------------------+
+|                                               | The sequences to apply the algorithm to.                            |
+|  - ``in_begin``, ``in_end``, ``out_begin`` (1)| Supported sequence types:                                           |
+|  - ``in_rng``, ``out_rng`` (2)                |                                                                     |
+|                                               | - ``sycl::buffer`` (2),                                             |
+|                                               | - :ref:`oneapi::dpl::experimental::ranges::views::all               |
+|                                               |   <viewable-ranges>` (2),                                           |
+|                                               | - :ref:`oneapi::dpl::experimental::ranges::views::subrange          |
+|                                               |   <viewable-ranges>` (2),                                           |
+|                                               | - :ref:`USM pointers <use-usm>` (1),                                |
+|                                               | - :ref:`oneapi::dpl::begin and oneapi::dpl::end                     |
+|                                               |   <use-buffer-wrappers>` (1).                                       |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+|  ``binary_op``                                | A function object that is applied to the elements of the input.     |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+|  ``param``                                    | A :doc:`kernel_param <../kernel_configuration>` object.             |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+
+
+**Type Requirements**:
+
+- The element type of sequence to scan must be a C++ integral or floating-point type
+  other than ``bool`` with a width of up to 64 bits.
```

Comment:
Is it correct? As I recall we test only 32-bit and 64-bit arithmetic types.

Reply:
You are right; we do not test 16-bit or 8-bit types so I updated this to make it clearer.